js/src/dropdown.js

```diff
@@ -257,6 +259,13 @@ class Dropdown extends BaseComponent {
 
     typeCheckConfig(NAME, config, this.constructor.DefaultType)
 
+    if (typeof config.reference === 'object' && !isElement(config.reference) &&
+      typeof config.reference.getBoundingClientRect !== 'function'
+    ) {
+      // Popper virtual elements require a getBoundingClientRect method
+      throw new Error(`${NAME}: Option "reference" provided type "object" without a required "getBoundingClientRect" method.`)
```

Comment:
Should we change this to a `TypeError` instead of the generic `Error`?

Reply:
True, missed it :smile: 
But then it should also be TypeError :thinking: 
https://github.com/twbs/bootstrap/blob/44667d89fa1bf8de15db57fad43696c62860e288/js/src/util/index.js#L119-L123

Reply:
OK, let's leave it as is for now and we tackle it later.